### PR TITLE
(FM-3903) Updates for release, pinning to 5.0.0 of the upstream module

### PR DESCRIPTION
**Pull request by garethr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,10 @@ To install Docker on a node, include the class `docker`.
 include 'docker'
 ~~~
 
-By default, for CentOS/RHEL, this installs Docker from your operating system's repo. For Ubuntu, this installs Docker from the upstream Docker repository.
+[Docker](https://github.com/docker/docker) from the [official
```
Comment (jbondpdx) [inline README.md]: Looks like the beginning of this sentence got lost. Just needs something like:

This installs...

File: README.md [hunk 2]
```diff
@@ -56,6 +59,32 @@ class {'docker':
 
 This example installs Docker version 0.5.5, binds the Docker daemon to a Unix socket and a tcp socket, provides the daemon with a dns server, and adds two users to the Docker group.
 
+Docker recently [launched new official
+repositories](http://blog.docker.com/2015/07/new-apt-and-yum-repos/#comment-247448)
+which are now the default for the module from version 5. If you want to
+stick with the old repositories you can do so with the following:
+
+```puppet
```
Comment (jbondpdx) [inline README.md]:
~~~ instead of ``

File: README.md [hunk 3]
```diff
@@ -249,6 +278,23 @@ Specifies extra mkfs arguments for creating the base device. Accepts a string.
 
 Specifies extra mount options for mounting thin devices. Accepts a string.
 
+#####`dm_thinpooldev
```
Comment (jbondpdx) [inline README.md]: missing closing backtick on dm_thinpooldev

File: README.md [hunk 4]
```diff
@@ -249,6 +278,23 @@ Specifies extra mkfs arguments for creating the base device. Accepts a string.
...
+
+Specifies a custom block storage device to use for the thin pool.
```
Comment (jbondpdx) [inline README.md]: Any default for this setting? Several of the added parameters look like they are missing this information; if they have no default, that's fine.

File: README.md [hunk 5]
```diff
@@ -249,6 +278,23 @@ Specifies extra mkfs arguments for creating the base device. Accepts a string.
...
+
+#####`dm_use_deferred_removal`
+
+Enables use of deferred device removal if libdm and the kernel driver support the mechanism.
+
+#####`dm_blkdiscard`
+
+Enables or disables the use of blkdiscard when removing devicemapper devices.
```
Comment (jbondpdx) [inline README.md]: Looks like maybe there's a hard carriage return that shouldn't be there. Also needs a period after Defaults to false.

File: README.md [hunk 6]
```diff
@@ -249,6 +278,23 @@ Specifies extra mkfs arguments for creating the base device. Accepts a string.
...
+Defaults to false
+
+#####`dm_override_udev_sync_check`
+
+By default, the devicemapper backend attempts to synchronize with the udev device manager for the Linux kernel. This option allows disabling that synchronization, to continue even though the configuration may be buggy. Defaults to true
```
Comment (jbondpdx) [inline README.md]: missing period at the end of default sentence

File: README.md [hunk 7]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
 #####`version`
 
 Specify a version of Docker to use. Valid options: a string containing a version number or 'latest'. Default: '0.5.5'.
 
+#####`log_driver`
+Set the log driver. Defaults to undef. Docker default is json-file.
+Valid values: none, json-file, syslog, journald, gelf, fluentd
+Valid values description:
+
+~~~
```
Comment (jbondpdx) [inline README.md]: I think this list of values and behavior should be a bulleted list rather than a code block, unless there's something special I'm missing?

File: README.md [hunk 8]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+     none     : Disables any logging for the container.
+                docker logs won't be available with this driver.
```
Comment (jbondpdx) [inline README.md]: Capitalize Docker

File: README.md [hunk 9]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+#####`storage_devs`
+A quoted, space-separated list of devices to be used.
+
+#####`storage_vg`
+The volume group to use for docker storage.
+
+#####`storage_root_size`
+The size to which the root filesystem should be grown.
+
+#####`storage_data_size`
```
Comment (jbondpdx) [inline README.md]: missing period; default?

File: README.md [hunk 10]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+The desired size for the docker data LV
+
+#####`storage_chunk_size`
```
Comment (jbondpdx) [inline README.md]: default?

File: README.md [hunk 11]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+Controls the chunk size/block size of thin pool.
+
+#####`storage_growpart`
```
Comment (jbondpdx) [inline README.md]: default?

File: README.md [hunk 12]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+Enable resizing partition table backing root volume group.
+
+#####`storage_auto_extend_pool`
+Enable/disable automatic pool extension using lvm
```
Comment (jbondpdx) [inline README.md]: missing period; default?

File: README.md [hunk 13]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+
+#####`storage_pool_autoextend_threshold`
```
Comment (jbondpdx) [inline README.md]: missing period

File: README.md [hunk 14]
```diff
@@ -498,6 +619,11 @@ Specifies whether to run the command in the background. Valid options: 'true', f
 
 Specifies whether to allocate a pseudo-TTY. Valid options: 'true', false'. Default: 'false'.
 
+
+#####`unless`
+A command to run in the context of the container, only if it returns a
```
Comment (jbondpdx) [inline README.md]: This was a little hard to parse. I'd suggest:

A command to run in the context of the container; if it returns a non-zero exit code, the command in the `command` is run.

I mean, assuming I've parsed it correctly, anyway. :-) Also, `command` should link back to the parameter it refers to.

File: README.md [hunk 15]
```diff
@@ -354,12 +400,87 @@ Specifies the tcp socket the Docker daemon should bind to. Accepts a string cont
...
+     gelf     : Graylog Extended Log Format (GELF) logging driver for Docker.
+                Writes log messages to a GELF endpoint: Graylog or Logstash.
+     fluentd  : Fluentd logging driver for Docker.
+                Writes log messages to fluentd (forward input).
+~~~
+
+#####`log_opt`
+Set the log driver specific options. Defaults to undef. Valid values per log driver:
+
+~~~
```
Comment (jbondpdx) [inline README.md]: Maybe this should be a bulleted list as well?  This list is a little more complicated, though, so I'm open to leaving it as a code block for the time being.

**Comment by bmjen** [comment]
Merged in #12
